### PR TITLE
[20765] Make RTPS reader implementations private

### PR DESCRIPTION
**Pull request by elianalf**
## Description

This PR makes RTPS reader implementations private:
- StatefulPersistentReader
- StatefulReader
- StatelessPersistentReader
- StatelessReader 

Related PR in Fast DDS Python: 

- https://github.com/eProsima/Fast-DDS-python/pull/120

## Contributor Checklist

- [x] Commit messages follow the project guidelines. 
- [x] The code follows the style guidelines of this project. 
- N/A Tests that thoroughly check the new feature have been added/Regression tests checking the bug and its fix have been added; the added tests pass locally 
- N/A Any new/modified methods have been properly documented using Doxygen. 
- N/A Any new configuration API has an equivalent XML API (with the corresponding XSD extension) 
- ❌ Changes are ABI compatible. 
- ❌ Changes are API compatible. 
- N/A New feature has been added to the `versions.md` file (if applicable).
- N/A New feature has been documented/Current behavior is correctly described in the documentation. 
    
- N/A Applicable backports have been included in the description.

## Reviewer Checklist

- [x] The PR has a milestone assigned.
- [x] The title and description correctly express the PR's purpose.
- [x] Check contributor checklist is correct.
- [ ] Check CI results: changes do not issue any warning.
- [ ] Check CI results: failing tests are unrelated with the changes.

### CODE REVIEW

**Comment by MiguelCompany** [comment]
@elianalf There are conflicts here, so a rebase is necessary.

File: src/cpp/rtps/reader/StatefulReader.h [hunk 1]
```diff
@@ -13,7 +13,7 @@
 // limitations under the License.
 
 /**
- * @file StatefulReader.h
+ * @file .h
```
Comment (MiguelCompany) [inline src/cpp/rtps/reader/StatefulReader.h:16]: Revert this

**Review by MiguelCompany — CHANGES_REQUESTED** [review]
It would be nice to also change the extension of the moved headers from `.h` to `.hpp

File: src/cpp/rtps/builtin/discovery/participant/PDPSimple.cpp [hunk 2]
```diff
@@ -44,7 +42,9 @@
 #include <rtps/builtin/liveliness/WLP.h>
 #include <rtps/history/TopicPayloadPoolRegistry.hpp>
 #include <rtps/participant/RTPSParticipantImpl.h>
+#include <rtps/reader/StatefulReader.hpp>
 #include <rtps/resources/TimedEvent.h>
+#include <rtps/reader/StatelessReader.hpp>
```
Comment (MiguelCompany) [inline src/cpp/rtps/builtin/discovery/participant/PDPSimple.cpp:47]:
```suggestion
#include <rtps/reader/StatefulReader.hpp>
#include <rtps/reader/StatelessReader.hpp>
#include <rtps/resources/TimedEvent.h>
```

File: src/cpp/rtps/participant/RTPSParticipantImpl.cpp [hunk 3]
```diff
@@ -64,6 +60,10 @@
 #include <rtps/network/utils/netmask_filter.hpp>
 #include <rtps/participant/RTPSParticipantImpl.h>
 #include <rtps/persistence/PersistenceService.h>
+#include <rtps/reader/StatefulPersistentReader.hpp>
+#include <rtps/reader/StatelessPersistentReader.hpp>
+#include <rtps/reader/StatefulReader.hpp>
+#include <rtps/reader/StatelessReader.hpp>
```
Comment (MiguelCompany) [inline src/cpp/rtps/participant/RTPSParticipantImpl.cpp:66]:
```suggestion
#include <rtps/reader/StatefulPersistentReader.hpp>
#include <rtps/reader/StatefulReader.hpp>
#include <rtps/reader/StatelessPersistentReader.hpp>
#include <rtps/reader/StatelessReader.hpp>
```

File: src/cpp/rtps/reader/StatefulPersistentReader.hpp [hunk 4]
```diff
@@ -13,15 +13,15 @@
 // limitations under the License.
 
 /**
- * @file StatefulPersistentReader.h
+ * @file StatefulPersistentReader.hpp
  */
 
 
 #ifndef _FASTDDS_RTPS_STATEFULPERSISTENTREADER_H_
```
Comment (MiguelCompany) [inline src/cpp/rtps/reader/StatefulPersistentReader.hpp:20]:
```suggestion
#ifndef RTPS_READER__STATEFULPERSISTENTREADER_HPP
```

File: src/cpp/rtps/reader/StatefulReader.hpp [hunk 5]
```diff
@@ -13,7 +13,7 @@
 // limitations under the License.
 
 /**
- * @file StatefulReader.h
+ * @file StatefulReader.hpp
  */
 
 #ifndef _FASTDDS_RTPS_READER_STATEFULREADER_H_
```
Comment (MiguelCompany) [inline src/cpp/rtps/reader/StatefulReader.hpp:19]:
```suggestion
#ifndef RTPS_READER__STATEFULREADER_HPP
```

File: src/cpp/rtps/reader/StatelessPersistentReader.hpp [hunk 6]
```diff
@@ -21,7 +21,7 @@
 #define _FASTDDS_RTPS_STATELESSPERSISTENTREADER_H_
```
Comment (MiguelCompany) [inline src/cpp/rtps/reader/StatelessPersistentReader.hpp:21]:
```suggestion
#define RTPS_READER__STATELESSPERSISTENTREADER_HPP
```

Comment (MiguelCompany) [reply src/cpp/rtps/reader/StatelessPersistentReader.hpp:21]: Change also comment in line 16.

File: src/cpp/rtps/reader/StatelessReader.hpp [hunk 7]
```diff
@@ -13,7 +13,7 @@
 // limitations under the License.
 
 /**
- * @file StatelessReader.h
+ * @file StatelessReader.hpp
  */
```
Comment (MiguelCompany) [inline src/cpp/rtps/reader/StatelessReader.hpp:19]: Change include guard to `RTPS_READER__STATELESSREADER_HPP`

**Review by MiguelCompany — CHANGES_REQUESTED** [review]

File: src/cpp/rtps/reader/StatefulReader.hpp (hunk 5 above)
Comment (MiguelCompany) [reply src/cpp/rtps/reader/StatefulReader.hpp:19]: Should also be applied to the mock header

File: src/cpp/rtps/reader/StatelessReader.hpp (hunk 7 above)
Comment (MiguelCompany) [reply src/cpp/rtps/reader/StatelessReader.hpp:19]: Should also be applied to the mock header

**Review by MiguelCompany — APPROVED** [review]

**Comment by elianalf** [comment]
@richiprosima please test_3 this

**Pull request merged**